cv-data-service-library/src/main/java/com/trihydro/library/helpers/TimGenerationHelper.java

```diff
@@ -272,25 +286,25 @@ private OdeTravelerInformationMessage updateTim(TimUpdateModel tum, OdeTravelerI
             // rebuild tim...this should maybe be extracted to another function and called
             // from multiple locations
             WydotTim wydotTim = getWydotTimFromTum(tum);
-            List<Milepost> mps = new ArrayList<>();
             List<Milepost> allMps = getAllMps(wydotTim);
             if (allMps.size() < 2) {
                 String exMsg = String.format(
-                        "Unable to resubmit TIM, less than 2 mileposts found to determine service area for Active_Tim %d",
-                        tum.getActiveTimId());
+                    "Unable to resubmit TIM, less than 2 mileposts found to determine service area for Active_Tim %d",
+                    tum.getActiveTimId());
                 utility.logWithDate(exMsg);
                 exceptions.add(new ResubmitTimException(tum.getActiveTimId(), exMsg));
                 return null;
             }
-            Milepost firstPoint = allMps.get(0);
-            Milepost secondPoint = allMps.get(1);
-
-            // reduce the mileposts by removing straight away posts
-            var anchorMp = getAnchorPoint(firstPoint, secondPoint);
-            mps = milepostReduction.applyMilepostReductionAlgorithm(allMps, config.getPathDistanceLimit());
+            Milepost anchorMp = attemptToCalculateAnchorPoint(tum, exceptions, allMps);
+            if (anchorMp == null) {
+                return null;
+            }
```

Comment:
Clarification (non-blocking): This part of the code now invokes the attemptToCalculateAnchorPoint method, which can handle the IdenticalPointsException.